### DISCUSSION TITLE
Switches and local binding- polling?

### DISCUSSION THREAD

**Discussion by jamesonuk** [Koenkk/zigbee2mqtt#19472] [Devices]
I am using some Hue Wall Switch modules with some Ikea bulbs and I am trying to play around with them to get them to work locally so the light switches will still work even if Z2M is down.
The instructions work fine (https://www.zigbee2mqtt.io/devices/929003017102.html#philips-929003017102) and I get local control but the issue is the group status gets out of sync.

Reading the binding docs (https://www.zigbee2mqtt.io/guide/usage/binding.html#state-changes) says 
> If your devices do not support reporting put the device in a group and bind the remote to the group instead of directly to the device. This will make Zigbee2MQTT poll the device for updates when the bound remote controls the device. To minimize traffic this has not been enabled for all devices. If this does not work please create an issue for it here.

and these devices do report (as do the bulbs they are controlling) but you can only bind them to groups.  Does this mean that updates are causing the bulbs to be polled?

The issues I am seeing (but I can't quite pinpoint an exact cause) is that HA is showing the lights as on or off when they are in fact the opposite.   This then causes an issue as when you trigger an update (via a switch or the app) it can take two or more attempts to actually change the state of the lights.

Ignoring the Hue specific stuff (when unbound this does appear to work as a true ZCL standard switch) I am left with PowerCfg and OnOff bindings.  With the device in the same group as the bulbs then
- with both bound to coordinator the local control works (lights toggle with switch) but the bulb status in Z2M still shows the old status and if I refresh they then update to the correct value
- with OnOff bound to group rather than coordinator (PowerCfg still to coordinator), Z2M configures reporting on the four bulbs in the group, I get local control and HA is in sync (when only using the switch, if I add the app in it gets out of sync again)

I can also bind OnOff to both coordinator and group but I am presuming that isn't good as you could have two things trying to update the same thing?

It seems that if I turn on / off solely via either the app or the switch it is generally OK but if I say turn it on via the switch and then off via the app this is when I start to see oddities a lot more often.

Is this triggering some sort of polling of the bubs in the group and they aren't reporting fast enough or is there something about how Z2M handles Hue devices (not sure if these switches work quite like the other Hue switches) or is there anything I can do to debug this?

An example of when the switch works was
```
Debug 2023-10-29 10:49:18Received Zigbee message from 'Study Hue Wall Switch', type 'commandToggle', cluster 'genOnOff', data '{}' from endpoint 1 with groupID 6
Info 2023-10-29 10:49:18MQTT publish: topic 'zigbee2mqtt/Study Hue Wall Switch', payload '{"action":"toggle","action_group":6,"battery":100,"device_mode":"single_rocker","last_seen":"2023-10-29T10:49:18.927Z","linkquality":255,"state":"OFF"}'
Info 2023-10-29 10:49:18MQTT publish: topic 'zigbee2mqtt/Study Hue Wall Switch', payload '{"action":"","battery":100,"device_mode":"single_rocker","last_seen":"2023-10-29T10:49:18.927Z","linkquality":255,"state":"OFF"}'
Info 2023-10-29 10:49:18MQTT publish: topic 'zigbee2mqtt/Study Hue Wall Switch/action', payload 'toggle'
Debug 2023-10-29 10:49:18Received Zigbee message from 'Study Door Right', type 'attributeReport', cluster 'genOnOff', data '{"onOff":1}' from endpoint 1 with groupID null
Info 2023-10-29 10:49:18MQTT publish: topic 'zigbee2mqtt/Study', payload '{"brightness":254,"color_mode":"color_temp","color_temp":250,"state":"ON"}'
Info 2023-10-29 10:49:18MQTT publish: topic 'zigbee2mqtt/Study Door Right', payload '{"brightness":254,"color_mode":"color_temp","color_options":{"execute_if_off":true},"color_temp":250,"color_temp_startup":65535,"last_seen":"2023-10-29T10:49:18.986Z","linkquality":255,"power_on_behavior":"previous","state":"ON","update":{"installed_version":65554,"latest_version":65554,"state":"idle"},"update_available":false}'
Debug 2023-10-29 10:49:18Received Zigbee message from 'Study Window Right', type 'attributeReport', cluster 'genOnOff', data '{"onOff":1}' from endpoint 1 with groupID null
Info 2023-10-29 10:49:18MQTT publish: topic 'zigbee2mqtt/Study Window Right', payload '{"brightness":254,"color_mode":"color_temp","color_options":{"execute_if_off":true},"color_temp":250,"color_temp_startup":65535,"last_seen":"2023-10-29T10:49:19.033Z","linkquality":255,"power_on_behavior":"previous","state":"ON","update":{"installed_version":65554,"latest_version":65554,"state":"idle"},"update_available":false}'
```
cut off the last bit but you can se it gets the toggle, messages are received from bulbs and then we publishes messages for the bulbs.

In the example below, the switch was toggled twice.  the lights responded and turned on then off but only one bulb seemed to send a message so HA was never updated
```
Debug 2023-10-29 10:49:18Received Zigbee message from 'Study Hue Wall Switch', type 'commandToggle', cluster 'genOnOff', data '{}' from endpoint 1 with groupID 6
Info 2023-10-29 10:49:18MQTT publish: topic 'zigbee2mqtt/Study Hue Wall Switch', payload '{"action":"toggle","action_group":6,"battery":100,"device_mode":"single_rocker","last_seen":"2023-10-29T10:49:18.927Z","linkquality":255,"state":"OFF"}'
Info 2023-10-29 10:49:18MQTT publish: topic 'zigbee2mqtt/Study Hue Wall Switch', payload '{"action":"","battery":100,"device_mode":"single_rocker","last_seen":"2023-10-29T10:49:18.927Z","linkquality":255,"state":"OFF"}'
Info 2023-10-29 10:49:18MQTT publish: topic 'zigbee2mqtt/Study Hue Wall Switch/action', payload 'toggle'
Debug 2023-10-29 10:49:18Received Zigbee message from 'Study Door Right', type 'attributeReport', cluster 'genOnOff', data '{"onOff":1}' from endpoint 1 with groupID null
Info 2023-10-29 10:49:18MQTT publish: topic 'zigbee2mqtt/Study', payload '{"brightness":254,"color_mode":"color_temp","color_temp":250,"state":"ON"}'
Info 2023-10-29 10:49:18MQTT publish: topic 'zigbee2mqtt/Study Door Right', payload '{"brightness":254,"color_mode":"color_temp","color_options":{"execute_if_off":true},"color_temp":250,"color_temp_startup":65535,"last_seen":"2023-10-29T10:49:18.986Z","linkquality":255,"power_on_behavior":"previous","state":"ON","update":{"installed_version":65554,"latest_version":65554,"state":"idle"},"update_available":false}'
```

**Comment by jamesonuk** [comment]
Bumping....
Does anyone actually  use binding?
Is this purely an issue with the Hue switch?

**Comment by chrisvandiemen** [comment]
@jamesonuk unfortunately I cannot help you with your question (I might run into the same problem later on) but I do have one for you.

I'm trying to bind my tradfri bulbs to my Hue wall switch, however I'm failing a the first step as I can't get my hue wall switch added to the z2m group (timeout error,  issue #19941 ).
Can you share some of the details of the environment you're running (Z2M version, adapter and adapter firmware)?

**Comment by jamesonuk** [comment]
> I'm trying to bind my tradfri bulbs to my Hue wall switch, however I'm failing a the first step as I can't get my hue wall switch added to the z2m group (timeout error, issue #19941 ).

@chrisvandiemen not to at home to check versions but you need to flick the switch to wake up the wall switch module (I do it a few times).   Then add it to the group and it should work.

**Comment by mundschenk-at** [comment]
As far as I know, even with the native Hue Bridge everything is controlled through the coordinator. It would be great if we had a reliable to use both direct binding for basic control and automations through the coordinator for more advanced stuff.
